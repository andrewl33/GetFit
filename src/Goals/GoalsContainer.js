import React, { Component } from 'react';
import { FlatList, StyleSheet, Text, View } from 'react-native';
import GoalListItem from './GoalListItem';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'stretch',
    marginTop: 50,
  },
  header: {
    fontSize: 35,
  },
  flatList: {
    alignSelf: 'stretch',
  },
});

export default class GoalsContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
    };
  }

  componentDidMount() {
    const fillItemList = [];
    for (let i = 0; i < 100; i++) {
      fillItemList.push({
        id: i.toString(),
        title: `Goal item ${i}`,
      });
    }

    this.setState({ items: fillItemList });
  }

  _keyExtractor = item => item.id;

  _renderItem = ({ item }) => <GoalListItem title={item.title} />;

  render() {
    const { items } = this.state;

    return (
      <View style={styles.container}>
        <Text style={styles.header}>GoalsContainer</Text>
        <FlatList
          style={styles.flatList}
          data={items}
          keyExtractor={this._keyExtractor}
          renderItem={this._renderItem}
        />
      </View>
    );
  }
}
