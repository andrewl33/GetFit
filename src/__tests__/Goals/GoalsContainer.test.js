import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import GoalsContainer from '../../Goals/GoalsContainer';

describe('GoalsContainer', () => {
  it('renders', () => {
    const tree = renderer.create(<GoalsContainer />);
    expect(tree).toMatchSnapshot();
  });
});
