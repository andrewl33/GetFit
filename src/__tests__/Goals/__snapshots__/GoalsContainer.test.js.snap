// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsContainer renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 35,
      }
    }
  >
    GoalsContainer
  </Text>
</View>
`;
